### PR TITLE
MR-516 - Publish AddRepairsContractsToAssetEvent after a New Asset is created

### PR DESCRIPTION
**Pull request by ange47rm**
Jira ticket - [MR-516](https://hackney.atlassian.net/jira/software/c/projects/MR/boards/88?modal=detail&selectedIssue=MR-516)

With this PR, after a new asset is created, and an SNS message of type `AssetCreatedEvent` is published, an additional message of type `AddRepairsContractsToAssetEvent` is also published (to the Asset SNS topic) if the `AddAssetRequest` has boolean property `AddDefaultSorContracts` set to `true`.

This message contains a GUID (which relates to the new asset's ID) and a property reference. The publishing of this message indicates that the property will need to have contracts added to it in the Repairs Hub database. This will be handled by the Repairs Listener.
image

Following this PR, we will have to set the Repairs Listener to listen for this message, so that it will run the stored procedure.

UPDATE: Repairs-Listener PR: https://github.com/LBHackney-IT/repairs-listener/pull/20

[MR-516]: https://hackney.atlassian.net/browse/MR-516?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: AssetInformationApi/V1/Infrastructure/AddRepairsContractsToAssetEventConstants.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace AssetInformationApi.V1.Infrastructure
+{
+    public static class AddRepairsContractsToAssetEventConstants
+    {
+        // JWT TOKEN
+        public const string V1_VERSION = "v1";
+        public const string EVENTTYPE = "AddRepairsContractsToAssetEvent";
+        public const string SOURCE_DOMAIN = "Asset";
```
Comment (ange47rm) [inline AssetInformationApi/V1/Infrastructure/AddRepairsContractsToAssetEventConstants.cs:8]: Is this SOURCE_DOMAIN correct? These properties were copied from the CreateAssetEvent, with the only difference being the EVENT_TYPE

File: AssetInformationApi/V1/UseCase/NewAssetUseCase.cs [hunk 2]
```diff
@@ -30,19 +32,31 @@ public NewAssetUseCase(IAssetGateway gateway, ISnsGateway snsGateway, ISnsFactor
         }
 
         [LogCall]
-        public async Task<AssetResponseObject> PostAsync(AssetDb request, Token token)
+        public async Task<AssetResponseObject> PostAsync(AddAssetRequest request, Token token)
```
Comment (ange47rm) [inline AssetInformationApi/V1/UseCase/NewAssetUseCase.cs:35]: Any refactoring suggestions on the above are welcome

File: AssetInformationApi/V1/Infrastructure/AddRepairsContractsToAssetEventConstants.cs (hunk 1 above)
Comment (Asura5) [reply AssetInformationApi/V1/Infrastructure/AddRepairsContractsToAssetEventConstants.cs:8]: This is correct

File: AssetInformationApi/V1/Infrastructure/AddRepairsContractsToNewAssetObject.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace AssetInformationApi.V1.Infrastructure
+{
+    // The same class is also defined in Repairs Listener to ensure the information sent and received is in the same format
+    public class AddRepairsContractsToNewAssetObject
+    {
+        public Guid EntityId { get; set; }
+        public string PropRef { get; set; }
+        public bool AddRepairsContracts { get; set; }
```
[the remark this answers is not in the document]
Comment (ange47rm) [reply AssetInformationApi/V1/Infrastructure/AddRepairsContractsToNewAssetObject.cs:10]: You're right, I can't see any service at the moment caring if the addContracts boolean is set to false. I see what you mean. So here in Asset API we could just add an if statement to check if the AddAssetRequest object has the boolean set to true, and if yes we send the AddContracts SNS message, just with the propRef (and entityId). I'll change it to that.

**Review by Asura5 — APPROVED** [review]

**Pull request merged**